### PR TITLE
Updates to cudnn package installation

### PR DESCRIPTION
**Pull request by perdasilva**
## Description ##
Currently, the installation script for centos7 cudnn only installs a single version of cudnn independendly of what is set in the CUDNN_VERSION environment variable. This PR aims to fix it by making it more general by constructing the download url for the cudnn package from the CUDA_VERSION and CUDNN_VERSION environment variables.

Additionally, by having the CUDNN_VERSION defined at the start of the Dockerfile, the whole docker file will be rebuilt if there's a change. Therefore, moving it next to the call to the install cudnn script.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [pr-awaiting-review, CI]

File: ci/docker/install/centos7_cudnn.sh [hunk 1]
```diff
@@ -22,11 +22,38 @@
 
 set -ex
 
+if [ -z ${CUDA_VERSION} ]; then
+    echo "Error: CUDA_VERSION environment variable undefiend"
```
Comment (stu1130) [inline ci/docker/install/centos7_cudnn.sh]: nit: undefined

File: ci/docker/install/centos7_cudnn.sh [hunk 2]
```diff
@@ -22,11 +22,38 @@
...
+    exit 1
+fi
+
+if [ -z ${CUDNN_VERSION} ]; then
+    echo "Error: CUDNN_VERSION environment variable undefiend"
```
Comment (stu1130) [inline ci/docker/install/centos7_cudnn.sh]: nit: undefined

Comment (perdasilva) [reply ci/docker/install/centos7_cudnn.sh]: Good eyes! It's fixed now.

**Comment by perdasilva** [comment]
@marcoabreu please review in light of our discussion

**Comment by wkcn** [comment]
Merged. Thank you!

**Pull request merged**